R/read_resource.R

```diff
@@ -199,6 +177,18 @@ read_resource <- function(package, resource_name) {
   schema <- get_schema(package, resource_name)
   fields <- schema$fields
 
+  # check if selected columns appear in schema
+  field_names <- purrr::map_chr(fields,~purrr::pluck(.x,"name"))
+  multiple_missing_columns <- sum(!col_select %in% field_names) > 1
+  assertthat::assert_that(
+    all(col_select %in% field_names),
+    msg = glue::glue(
+      "Can't find {ifelse(multiple_missing_columns,'columns','column')} ",
```

Comment:
I would just write `column(s)` here and drop the functionality to assess singular from plural.